modules/ROOT/nav.adoc

```diff
@@ -37,6 +37,7 @@
 * xref:cdi-beans.adoc[Context and Dependency Injection]
   ** xref:jaxrs-integration-cdi.adoc[RESTful Web Services integration with CDI]
   ** https://openliberty.io/guides/cdi-intro.html[Guide: Injecting dependencies into microservices]
+  ** xref:cdi-extension-user-feature.adoc[Creating a CDI Extension inside a User Feature]
```

Comment:
to avoid branch conflicts between draft-staging-production, we only update the nav file from a dedicated branch:

https://github.com/OpenLiberty/docs?tab=readme-ov-file#editing-the-docs-navigation

I went ahead and added this entry from the `draft-nav` branch in #7532